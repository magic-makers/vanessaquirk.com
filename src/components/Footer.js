import React from 'react';
import config from '../../config';
export default function Footer() {
  return (
    <footer id="footer">
      <div className="inner">
        <section>
          <h2>Get in touch</h2>
          <ul className="icons">
            {config.socialLinks.map(social => {
              const { icon, name, url } = social;
              return (
                <li key={url}>
                  <a
                    href={url}
                    target="_blank"
                    rel="noopener noreferrer"
                    className={`icon fa-2x ${icon}`}
                  >
                    <span className="label">{name}</span>
                  </a>
                </li>
              );
            })}
          </ul>
        </section>
      </div>
    </footer>
  );
}
