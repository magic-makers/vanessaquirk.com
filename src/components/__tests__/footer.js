import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../Footer';

jest.mock('../../../config', () => ({
  socialLinks: [
    {
      icon: 'icon-1',
      name: 'Link 1',
      url: 'https://foo.com',
    },
    {
      icon: 'icon-2',
      name: 'Link 2',
      url: 'https://bar.com',
    },
  ],
}));

describe('Footer', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
