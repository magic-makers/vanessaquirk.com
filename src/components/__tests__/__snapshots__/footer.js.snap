// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  id="footer"
>
  <div
    className="inner"
  >
    <section>
      <h2>
        Get in touch
      </h2>
      <form
        action="#"
        method="post"
      >
        <div
          className="fields"
        >
          <div
            className="field half"
          >
            <input
              id="name"
              name="name"
              placeholder="Name"
              type="text"
            />
          </div>
          <div
            className="field half"
          >
            <input
              id="email"
              name="email"
              placeholder="Email"
              type="email"
            />
          </div>
          <div
            className="field"
          >
            <textarea
              id="message"
              name="message"
              placeholder="Message"
            />
          </div>
        </div>
        <ul
          className="actions"
        >
          <li>
            <input
              className="primary"
              type="submit"
              value="Send"
            />
          </li>
        </ul>
      </form>
    </section>
    <section>
      <h2>
        Follow
      </h2>
      <ul
        className="icons"
      >
        <li>
          <a
            className="icon icon-1"
            href="https://foo.com"
          >
            <span
              className="label"
            >
              Link 1
            </span>
          </a>
        </li>
        <li>
          <a
            className="icon icon-2"
            href="https://bar.com"
          >
            <span
              className="label"
            >
              Link 2
            </span>
          </a>
        </li>
      </ul>
    </section>
    <ul
      className="copyright"
    >
      <li>
        Â© Phantom. All rights reserved
      </li>
      <li>
        Design: 
        <a
          href="http://html5up.net"
        >
          HTML5 UP
        </a>
      </li>
    </ul>
  </div>
</footer>
`;
